### PR TITLE
Feature/6636 dynamic clean up

### PR DESCRIPTION
**Pull request by assoomro**
AB#6636
AB#7662

### CODE REVIEW

File: tear-down-azure.sh [hunk 1]
```diff
@@ -1,13 +1,101 @@
 #!/bin/bash
 
-if [ $# -ne 1  ]
-then
-  echo "usage ./launch-or-update-azure.sh <namespace>"
+function displayHelp {
+  printf "usage: ./tear-down-azure.sh [OPTIONS]
```
Comment (stuart-bass-bjss) [inline tear-down-azure.sh:4]: Can you please make it so that this still works given a namespace name?

File: tear-down-azure.sh [hunk 2]
```diff
@@ -1,13 +1,146 @@
...
+    if [ ! $1 ]; then
+        branchNamespace=`echo $branchName | sed 's/feature[[:punct:]]/bc-/g'`
+        prNamespace="bc-merge-$prNumber"
+        return 
+    fi
+    
+    branchNamespace="$1"
+}
+
+function deleteKuberetesResources {
```
Comment (stuart-bass-bjss) [inline tear-down-azure.sh:99]: Kubernetes

Comment (assoomro) [reply tear-down-azure.sh:99]: 😄 Thanks!

File: tear-down-azure.sh [hunk 3]
```diff
@@ -1,13 +1,146 @@
 #!/bin/bash
```
Comment (stuart-bass-bjss) [inline tear-down-azure.sh:1]: Just running this crashes in an interesting way. Also can't add -n as instructed

Comment (assoomro) [reply tear-down-azure.sh:1]: What seems to be the issue?

File: tear-down-azure.sh [hunk 4]
```diff
@@ -1,13 +1,146 @@
...
+
+if [ -z "$commitMessage" ]; then
+    echo "missing commit message argument, relying on provided namespace name..."
+    calculateNamespaceNames "$namespace"
+    deleteAllResources
+    exit
 fi
 
-namespace=$1
+if [ -z "$commitMessage" ] && [ -z "$namespace" ]; then
```
Comment (stuart-bass-bjss) [inline tear-down-azure.sh:137]: Needs to also check the other required parameters

File: tear-down-azure.sh [hunk 5]
```diff
@@ -1,13 +1,146 @@
...
+    services=("bapi"  "isapi"  "ordapi")
+    deleteQueries=()
+
+    for service in ${services[*]}; do
+        deleteQueries+=("DROP DATABASE [bc-$branchNamespace-$service];")
+        deleteQueries+=("DROP DATABASE [bc-$prNamespace-$service];")
+    done
+
+    for query in ${deleteQueries[*]}; do
+        sqlcmd -S $dbServer -U "$saUserName" -P "$saPassword" -d master -Q $query || true
```
Comment (stuart-bass-bjss) [inline tear-down-azure.sh:120]: Can you use `az sql db delete` instead? Otherwise, need to install sqlcommand.

**Pull request merged**